src/use-snap-carousel.tsx

```diff
@@ -270,20 +273,36 @@ const getScrollPaddingUsedValue = (el: HTMLElement, pos: 'left' | 'top') => {
     assert(!Number.isNaN(value), invalidMsg);
     return (el.clientWidth / 100) * value;
   }
+  // e.g. calc(10% + 10px) // NOTE: We could in theory resolve this...
   throw new RSCError(invalidMsg);
 };
 
-// The 'effective' scroll padding is the actual padding that will be used for a specific item
-// after factoring in whether there is enough scroll width available.
-// TODO: Consider factoring in an item's scroll-margin too.
-const getEffectiveScrollPadding = (
+// Unlike scroll-padding, scroll-margin doesn't support <percentage> values, so we should always
+// get back a px value, meaning it's effectively already the *used* value.
+// https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-margin
+const getScrollMarginUsedValue = (el: HTMLElement, pos: 'left' | 'top') => {
+  const style = window.getComputedStyle(el);
+  const scrollMargin = style.getPropertyValue(`scroll-margin-${pos}`);
+  // https://developer.mozilla.org/en-US/docs/Web/CSS/length
+  // https://www.w3.org/TR/css3-values/#length-value
+  const invalidMsg = `Unsupported scroll margin value, expected <length> value, received ${scrollMargin}`;
+  assert(scrollMargin.endsWith('px'), invalidMsg); // Even scroll-margin: 0 should return "0px"
+  const value = parseInt(scrollMargin);
+  assert(!Number.isNaN(value), invalidMsg);
+  return value;
+};
```

Comment:
TBH if my understanding of the spec is correct, the value should always come back as `px` so no need to validate here but being thorough to catch any misunderstandings.